src/App.vue

```diff
@@ -39,29 +39,53 @@
 export default {
   data: () => ({
     lists:[],
-    newTask:''
+    newTask:'',
+    displayState:''
   }),
+  computed:{
+    viewTodos(){
+      return this.displayState
+      ? this.lists.filter((i) => i.status === this.displayState)
+      : this.lists
+    }
+  },
   methods:{
     // Todoの追加
     addList:function(e){
       e.preventDefault();
       if (this.newTask.match(/\S/g)) 
         this.lists.push({
+          id: this.lists.length,
           task: this.newTask,
           status:"作業中",
       });
+      this.newTask=''
     },
     //削除機能
-    deleteList:function(key){
-      this.lists.splice(key, 1)
+    deleteList:function(id){
+      if(id > -1)
+      this.lists.splice(id, 1);
+      this.lists.forEach((list, key) => {
+        list.id = key
+      })
     },
     //状態の変更
-    changeStatus:function(key) {
-      if (this.lists[key].status === "作業中"){
-        this.lists[key].status = "完了";
-      }else if(this.lists[key].status === "完了") {
-        this.lists[key].status = "作業中";
+    changeStatus:function(id){
```

Comment:
functionの記述は基本的には省略してしまって良いですが、省略するならする、しないならしないで統一しましょう。
```suggestion
    changeStatus(id) {
```